### PR TITLE
refactor: Centralise all retrieval and storage of user information

### PR DESCRIPTION
**Pull request by john-gom**
Make sure you've done all the following
- [ ] Include unit tests for new functionality
- [ ] Code passes GitHub workflow checks in your branch
- [ ] If you have multiple commits please combine them into one commit by squashing them.

### What

Move all user storage and retrieval to a single place so it is easier to switch to fetching data from Keycloak

### Related issue(s) and discussion
- Related to #1596

### CODE REVIEW

File: lib/ProductOpener/Missions.pm [hunk 1]
```diff
@@ -150,27 +150,17 @@ sub gen_missions_html() {
 
 sub compute_missions() {
 
-	opendir DH, $BASE_DIRS{USERS} or die "Couldn't open the current directory: $!";
-	my @userids = sort(readdir(DH));
-	closedir(DH);
+	my @userids = retreive_userids();
```
Comment (stephanegigandet) [inline lib/ProductOpener/Missions.pm:153]:
```suggestion
	my @userids = retrieve_userids();
```

File: lib/ProductOpener/Users.pm [hunk 2]
```diff
@@ -960,25 +937,97 @@ sub retrieve_user ($user_id) {
...
+
+	if ((defined $email) and ($email =~ /\@/)) {
+		if (defined $emails_ref->{$email}) {
+			delete $emails_ref->{$email};
+			store("$BASE_DIRS{USERS}/users_emails.sto", $emails_ref);
 		}
 	}
+}
+
+sub retreive_userids() {
```
Comment (stephanegigandet) [inline lib/ProductOpener/Users.pm:1010]:
```suggestion
sub retrieve_userids() {
```

**Review by stephanegigandet — APPROVED** [review]
Looks good to me, thank you!

File: tests/integration/create_pro_user.t [hunk 3]
```diff
@@ -54,6 +54,7 @@ ok(!html_displays_error($resp), "no error creating pro user");
 my $logs = tail_log_read($tail);
 
 # As it is the first user of the org, user is already part of the org
+# TODO: This may need to chane with Keycloak
```
Comment (teolemon) [inline tests/integration/create_pro_user.t:57]:
```suggestion
# TODO: This may need to change with Keycloak
```

**Pull request merged**